python/arcticdb/version_store/processing.py

```diff
@@ -489,10 +500,20 @@ def agg(self, aggregations: Dict[str, str]):
             len(self.clauses) and isinstance(self.clauses[-1], _GroupByClause),
             f"Aggregation only makes sense after groupby",
         )
+        group_by_clause = self.clauses[-1]
+
         for v in aggregations.values():
             v = v.lower()
-        self.clauses.append(_AggregationClause(self.clauses[-1].grouping_column, aggregations))
+        self.clauses.append(_AggregationClause(group_by_clause.grouping_column, aggregations))
         self._python_clauses.append(PythonAggregationClause(aggregations))
+
+        if group_by_clause.sort:
+            self.clauses.append(_MergeClause())
```

Comment:
Isn't this the wrong way round? The sort clause sorts the individual segments, so that we can merge them together.

Reply:
Merging individually sorted segments does not result in a merged segment, hence the current order.